packages/firestore/lite/src/api/transaction.ts

```diff
@@ -123,9 +123,14 @@ export class Transaction {
    *
    * @param documentRef A reference to the document to be set.
    * @param data An object of the fields and values for the document.
+   * @param options An object to configure the set behavior.
    * @return This `Transaction` instance. Used for chaining method calls.
    */
-  set<T>(documentRef: DocumentReference<T>, data: Partial<T>): this;
+  set<T>(
+    documentRef: DocumentReference<T>,
+    data: Partial<T>,
+    options: SetOptions
```

Comment:
This somehow doesn't compile in this PR, so I had to fix this. This likely broke when I merged https://github.com/firebase/firebase-js-sdk/pull/3843, but I don't know how I was able to merge it.